src/system-program.js

```diff
@@ -517,14 +538,29 @@ export class SystemProgram {
   /**
    * Generate a Transaction that creates a new Nonce account
    */
-  static createNonceAccount(params: CreateNonceAccountParams): Transaction {
-    let transaction = SystemProgram.createAccount({
-      fromPubkey: params.fromPubkey,
-      newAccountPubkey: params.noncePubkey,
-      lamports: params.lamports,
-      space: NONCE_ACCOUNT_LENGTH,
-      programId: this.programId,
-    });
+  static createNonceAccount(
+    params: CreateNonceAccountParams | CreateNonceAccountWithSeedParams,
+  ): Transaction {
+    let transaction;
+    if (params.basePubkey && params.seed) {
```

Comment:
Since it's cheap to do here, I think we should do the check that `seed` serializes into a buffer of no more than 32 bytes client-side.

Reply:
Sounds good! Created a new issue :) we need to apply this to create account as well